### PR TITLE
Wire RefreshView up to our xplat layout workflow

### LINKED ISSUE #23029: Regression Issue: CollectionView SizeChanged does not work in SR6 but it works in SR5  (.NET 8)
### Description

I have a CollectionView and a registered behavior called 'ResponsiveGridBehavior'. The behavior hooks the 'SizeChanged' event to control the number of span columns. The behavior is intended to work only on Desktop Idiom like Windows, Mac Catalyst

```csharp
public class ResponsiveGridBehavior : Behavior<CollectionView>
{
	protected override void OnAttachedTo(CollectionView bindable)
	{
		base.OnAttachedTo(bindable);
		if (DeviceInfo.Idiom == DeviceIdiom.Desktop)
			bindable.SizeChanged += OnSizeChanged;
	}

	protected override void OnDetachingFrom(CollectionView bindable)
	{
		if (DeviceInfo.Idiom == DeviceIdiom.Desktop)
			bindable.SizeChanged -= OnSizeChanged;
		base.OnDetachingFrom(bindable);
	}

	private void OnSizeChanged(object? sender, EventArgs e)
	{
		if (sender is CollectionView collectionView && collectionView.ItemsLayout is GridItemsLayout gridItemsLayout)
		{
			double maxWidthPerItem = Application.Current!.GetResource<double>("CardMaximumWidth", 300);
			gridItemsLayout.Span = Math.Max(1, (int)(collectionView.Width / maxWidthPerItem));
		}
	}
}
```

After updating to SR6 the event 'SizeChanged' is never called.

### Steps to Reproduce

_No response_

### Link to public reproduction project repository

https://github.com/malsabi/CollectionViewSizeChangedIssue

### Version with bug

Unknown/Other

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

8.0.40 SR5

### Affected platforms

Windows, macOS

### Affected platform versions

_No response_

### Did you find any workaround?

I had to roll back to SR5.

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change

`RefreshView` inherits from `ContentView` so when it measures and arranges its children it needs to use the layout flow expected by `ContentView` otherwise things like `Frame` won't get wired up during the first `Arrange` pass

### Issues Fixed

Fixes #23029

### CODE REVIEW

File: src/Core/src/Platform/iOS/MauiRefreshView.cs [hunk 1]
```diff
@@ -166,17 +168,6 @@ bool TryInsertRefresh(UIView view, int index = 0)
...
 		}
 
-		public override CGRect Bounds
-		{
-			get => base.Bounds;
-			set
-			{
-				base.Bounds = value;
-				if (_contentView != null)
-					_contentView.Frame = value;
```
Comment (PureWeen) [inline src/Core/src/Platform/iOS/MauiRefreshView.cs:176]: This was most likely here to workaround the fact that we didn't have the layout path setup correctly. 

It looks like the original addition of this didn't add tests
https://github.com/dotnet/maui/pull/14302

So, it'd be good to test through those closed issues

Comment (PureWeen) [reply src/Core/src/Platform/iOS/MauiRefreshView.cs:176]: I ran the sample [here](https://github.com/dotnet/maui/issues/11327) and it looks good still

image

**Comment by Redth** [comment]
Tested with the PR build packages on iOS, MacCatalyst, and Windows (touch screen).  I tested functionality before these changes and after.

Working well in every platform I tested.  Windows is a bit odd of behaviour (it requires a scrollview inside it, and it requires the content to need to be scrolled for pull to refresh to work), but it is no different than before this PR.

**Review by Redth — APPROVED** [review]

**Review by Redth — APPROVED** [review]

**Comment by Redth** [comment]
Tested again after the last changes too, things seem to be working fine still.

**Pull request merged**